### ISSUE TITLE
Zend\Expressive\Delegate\DefaultDelegate missing

### ISSUE THREAD

**Issue by RalfEggert** [zendframework/zend-expressive#465]
After using the `expressive-pipeline-from-config` Skript the file `/config/autoload/programmatic-pipeline.global.php/ is created. In this file the class `Zend\Expressive\Delegate\DefaultDelegate` is used but this class does not exist. 

Bug or feature?

**Comment by xtreamwayz** [comment]
Sort of a feature.

The [ApplicationFactory](https://github.com/zendframework/zend-expressive/blob/39884d99686b71308b965876e632bc6ab8ca3203/src/Container/ApplicationFactory.php#L69-L71) checks for it. The class doesn't need to exist. For PHP it is just a string as long as it has the `::class` extension. I think in Expressive 1 the same method was used for the FinalHandler.

**Comment by RalfEggert** [comment]
In ZE 1 we used:

```
'Zend\Expressive\FinalHandler' => TemplatedErrorHandlerFactory::class,
```

In ZE 2 the generated code contains:

```
DefaultDelegate::class => NotFoundDelegate::class,
```

And this class does not exist.

**Comment by xtreamwayz** [comment]
It probably was another class than. At the time I made the same comment and learned from @weierophinney that it can be done this way.

**Comment by xtreamwayz** [comment]
As an [example](https://3v4l.org/KTBeE):

```php
<?php

var_dump(
    Zend\Expressive\Delegate\DefaultDelegate::class,
    'Zend\Expressive\Delegate\DefaultDelegate'
);
```

Output:

```
string(40) "Zend\Expressive\Delegate\DefaultDelegate"
string(40) "Zend\Expressive\Delegate\DefaultDelegate"
```

**Comment by froschdesign** [comment]
@RalfEggert 
> And this class does not exist.

In PHP the class does not have to exist or be loaded.

@xtreamwayz 
> … and learned from @weierophinney that it can be done this way.

Is this a good way?  I think no!

- Using the class name but the class doesn't exists is confusing. [And using this in a "Getting Started" tutorial it's a no go.](https://github.com/zendframework/tutorials/issues/74)
- Creating test doubles will not work.
- An IDE can not support this for autocompletion or inspection.
- And do not forget this issue itself!

**Comment by RalfEggert** [comment]
I agree with @froschdesign, it just raises more questions than it solves. If the class does not exist, please don't use the `::class` constant and just use a plain old string. It is confusing for beginners.

**Comment by weierophinney** [comment]
Could one of you create PRs for:

- zendframework/zend-expressive-skeleton to change the DefaultDelegate entry in the configuration
- zendframework/zend-expressive to update the documentation
- zendframework/zend-expressive-tooling to update the "expressive-pipeline-from-config" tool to not emit `::class` notation when generating configuration for the delegate

I can see the arguments for why this usage is confusing, and those are the places that would need to change.

Thanks in advance!

**Comment by xtreamwayz** [comment]
/ping @weierophinney 

I think I've changed all references.

**Comment by weierophinney** [comment]
I've merged each of these PRs, and released zend-expressive-tooling 0.3.2. I'll likely release zend-expressive 2.0.3 today; there are a number of pending PRs on the skeleton, however, so I'm not sure how quickly I'll get to those. Once those are complete, however, I'll release 2.0.1.

**Comment by froschdesign** [comment]
@xtreamwayz 
Thanks for your PRs! 👍 

@weierophinney 
Thanks for your quick response on this topic! 👍

**Comment by RalfEggert** [comment]
Nice job to all of you! :-)

**Comment by froschdesign** [comment]
Fixed with #468 and released with version 2.0.3.
